W-17821960-metrics-api-duke (#347)
mulesoft/docs-monitoring: modules/ROOT/pages/anypoint-insights.adoc

+
{selector-all-dashboards}

> Anypoint Insights uses <<metrics-api, Anypoint Monitoring Metrics API>> to query data for charts. Data aggregation and retention limits in Anypoint Insights are derived from the service that supports the Metrics API. 

Author: ```suggestion
Anypoint Insights uses <<metrics-api, Anypoint Monitoring Metrics API>> to query for data that appears in charts. Data aggregation and retention limits in Anypoint Insights are derived from the service that supports the Metrics API. 
```

> Anypoint Insights uses <<metrics-api, Anypoint Monitoring Metrics API>> to collect data for charts. Data aggregation and retention limits in Anypoint Insights are derived from the service that supports the Metrics API. 

Author: ```suggestion
Anypoint Insights uses <<metrics-api, Anypoint Monitoring Metrics API>> to query data for charts. Data aggregation and retention limits in Anypoint Insights are derived from the service that supports the Metrics API. 
```

[[automatic-interval]]
=== Data Points in Anypoint Monitoring Charts

> Anypoint Monitoring charts aggregate data into data points along a horizontal *Time* axis. If the units time for data in your charts are larger or smaller than you require, try narrowing or expanding the time range that your charts capture. Alternatively, you can use the <<metrics-api, Metrics API>> to aggregate data within the API's limits.

Author: @sathishpaulleo Could you look at ☝️

Reviewer: Looks good to me.

> Anypoint Monitoring charts aggregate data into data points along a horizontal *Time* axis. If the units time for data in your charts are larger or smaller than you require, try narrowing or expanding the time range that your charts capture. Alternatively, you can use the <<metrics-api, Metrics API>> to aggregate data within the API's limits.

Author: ```suggestion
Anypoint Monitoring charts aggregate data into data points along a horizontal *Time* axis. If the time units for data in your charts are larger or smaller than you require, try narrowing or expanding the time range that your charts capture. Alternatively, you can use the <<metrics-api, Metrics API>> to aggregate data within the API's limits.
```

//include for How Anypoint Monitoring Collects Metrics
[[data-collection-intervals]]
include::partial$include-monitoring-metrics-collection.adoc[leveloffset=+2]

[[metrics-api]]
== Metrics API

Use Anypoint Monitoring Metrics API to access metrics that Anypoint Monitoring exposes. The documentation for this API is available on Anypoint Exchange. 

> //TODO: See ADD_URL_TO_API_ON PROD EXCHANGE_ 

Author: ```suggestion
See: https://anypoint.mulesoft.com/exchange/TODO_TODO
```
Need URL or we can perhaps point to a search for the API if we don't have this in time.

> //TODO: QAX URL (don't use): https://qax.anypoint.mulesoft.com/exchange/6046b96d-c9aa-4cb2-9b30-90a54fc01a7b/metrics-api/minor/1.0/pages/home/ 

Author: ```suggestion
```
remove after adding new URL

[[automatic-interval]]
=== Data Points in Anypoint Monitoring Charts

Anypoint Monitoring charts aggregate data into data points along a horizontal *Time* axis. If the time units for data in your charts are larger or smaller than you require, try narrowing or expanding the time range that your charts capture. Alternatively, you can use the <<metrics-api, Metrics API>> to aggregate data within the API's limits.

//include for How Anypoint Monitoring Collects Metrics
[[data-collection-intervals]]
include::partial$include-monitoring-metrics-collection.adoc[leveloffset=+2]

[[metrics-api]]
== Metrics API

> Anypoint Insights uses Anypoint Monitoring Metrics API to collect data for charts. You can use this API to query data for your needs. 

Author: ```suggestion
Anypoint Insights uses Anypoint Monitoring Metrics API to submit queries for data used in charts. You can use this API to query data for your needs. 
```

> Anypoint Insights uses Anypoint Monitoring Metrics API to collect data for charts. You can use this API to collect metrics data for your needs. 

Reviewer 2: should this be "query data"? IMO, Collection usually refers to metrics being pulled from actual source.

> Anypoint Insights uses Anypoint Monitoring Metrics API to collect data for charts. You can use this API to collect metrics data for your needs. 

Author: ```suggestion
Anypoint Insights uses Anypoint Monitoring Metrics API to collect data for charts. You can use this API to query data for your needs. 
```

Age-based data aggregation limits determine the minimum intervals available for retrieving metrics data:

* Data less than or equal to 7 days old is available at a minimum of 1-minute intervals. 
* Data over 7 days old is available at a minimum of 5-minute intervals. 

> To use the API and get information about data retention, see the API documentation on Anypoint Exchange: https://anypoint.mulesoft.com/exchange/f1e97bc6-315a-4490-82a7-23abe036327a.anypoint-platform/metrics-api/minor/1.0/ 

Author: ```suggestion
To use the API and get information about data retention, see the API documentation on Anypoint Exchange: https://anypoint.mulesoft.com/exchange/f1e97bc6-315a-4490-82a7-23abe036327a.anypoint-platform/metrics-api/minor/1.0/[Anypoint Monitoring Metrics API]
```

[[automatic-interval]]
=== Data Points in Anypoint Monitoring Charts

Anypoint Monitoring charts aggregate data into data points along a horizontal *Time* axis. If the time units for data in your charts are larger or smaller than you require, try narrowing or expanding the time range that your charts capture. Alternatively, you can use the <<metrics-api, Metrics API>> to aggregate data within the API's limits.

//include for How Anypoint Monitoring Collects Metrics
[[data-collection-intervals]]
include::partial$include-monitoring-metrics-collection.adoc[leveloffset=+2]

[[metrics-api]]
== Metrics API

Anypoint Insights uses the Metrics API to collect data for charts. You can use this API to collect metrics data for your needs. 

> Data aggregation periods depend on the age of the data:

Author: ```suggestion
Age-based data aggregation limits determine the intervals available for retrieving metrics data:
```

> Age-based data aggregation limits determine the intervals available for retrieving metrics data:

Author: ```suggestion
Age-based data aggregation limits determine the minimum intervals available for retrieving metrics data:
```

[[automatic-interval]]
=== Data Points in Anypoint Monitoring Charts

> Anypoint Monitoring charts aggregate data into data points along a horizontal *Time* axis. If the time units for data in your charts are larger or smaller than you require, try narrowing or expanding the time range that your charts capture. Alternatively, you can use the <<metrics-api, Metrics API>> to aggregate data within the API's limits.

Author: This part was approved by Sathish.

//include for How Anypoint Monitoring Collects Metrics
[[data-collection-intervals]]
include::partial$include-monitoring-metrics-collection.adoc[leveloffset=+2]

[[metrics-api]]
== Metrics API

Anypoint Insights uses the Metrics API to collect data for charts. You can use this API to collect metrics data for your needs. 

> Data aggregation periods depends on the age of the data:

Author: ```suggestion
Data aggregation periods depend on the age of the data:
```

* Data less than or equal to 7 days old is aggregated into 1-minute segments. 
* Data over 7 days old is aggregated into 5-minute segments. 

> //TODO: FILL IN TODO IN URL

Author: ```suggestion
```

> To use the API and get information about data retention, see the API documentation on Anypoint Exchange: https://anypoint.mulesoft.com/exchange/TODO_TODO

Author: TODO: COMPLETE URL BEFORE MERGING - search query added for now until we know precise URL
```suggestion
To use the API and get information about data retention, see the API documentation on Anypoint Exchange: https://anypoint.mulesoft.com/exchange/f1e97bc6-315a-4490-82a7-23abe036327a.anypoint-platform/metrics-api/minor/1.0/ 
```

Reviewer 2: Use:
https://anypoint.mulesoft.com/exchange/f1e97bc6-315a-4490-82a7-23abe036327a.anypoint-platform/metrics-api/minor/1.0/

//TODO: QAX URL (don't use): https://qax.anypoint.mulesoft.com/exchange/6046b96d-c9aa-4cb2-9b30-90a54fc01a7b/metrics-api/minor/1.0/pages/home/ //

> //TODO: IF URL UNAVAILABLE, perhaps we could just put search URL https://anypoint.mulesoft.com/exchange/?search=%27Anypoint%20Monitoring%20Metrics%20API%27&show=all-org&sort=score Should be the first hit once API is up on the site.

Author: TO REMOVE BEFORE MERGING:
```suggestion
```

Author: It may make sense for me to also add data retention info here in our doc if that is completely settled by 10am Thurs. Or we can add it later.
